pkg/dogstatsd/filter/filter.go

```diff
@@ -0,0 +1,62 @@
+package filter
+
+import (
+	"regexp"
+	"strings"
+)
+
+type TagFilter struct {
+	filters []*regexp.Regexp
+}
+
+func NewTagFilter(regexes []string) (*TagFilter, error) {
+	filters := make([]*regexp.Regexp, len(regexes))
+	for i, r := range regexes {
+		regex, err := regexp.Compile(r)
+		if err != nil {
+			return nil, err
+		}
+
+		filters[i] = regex
+	}
+	return &TagFilter{
+		filters: filters,
+	}, nil
+}
+
+func (tf *TagFilter) Filter(tags []string) []string {
+	var i int
+	for _, t := range tags {
+		if ok, result := tf.filter(t); ok {
+			tags[i] = result
+			i++
+		}
+	}
+
+	// Prevent memory leak by erasing truncated values
```

Comment:
This is only needed if the slice contains pointer types. The underlying array isn't going to change size even though we're returning a new, more limited slice. This could reduce the total size of the data contained in the array by shifting non-zero values to "" but there's not really a chance of a leak here in the sense of one that would be solved with zero values.